crates/router/src/connector/expresscheckout.rs

```diff
@@ -0,0 +1,618 @@
+        ];
+        let auth = expresscheckout::ExpresscheckoutAuthType::try_from(&req.connector_auth_type)?;
+        let auth_header = (
+            headers::AUTHORIZATION.to_string(),
+            format!("Basic {}", auth.api_key),
+        );
+        headers.push(auth_header);
+        Ok(headers)
+    }
+}
+
+impl ConnectorCommon for Expresscheckout {
+    fn id(&self) -> &'static str {
+        "expresscheckout"
+    }
+
+    fn common_get_content_type(&self) -> &'static str {
+        "application/x-www-form-urlencoded"
+    }
+
+    fn base_url<'a>(&self, connectors: &'a settings::Connectors) -> &'a str {
+        connectors.expresscheckout.base_url.as_ref()
+    }
+
+    // fn get_auth_header(&self, auth_type:&types::ConnectorAuthType)-> CustomResult<Vec<(String,String)>,errors::ConnectorError> {
+    //     let auth: expresscheckout::ExpresscheckoutAuthType = auth_type
+    //         .try_into()
+    //         .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+    //     Ok(vec![(headers::AUTHORIZATION.to_string(), auth.api_key)])
+    // }
+}
+
+impl api::Payment for Expresscheckout {}
+
+impl api::PreVerify for Expresscheckout {}
+impl ConnectorIntegration<api::Verify, types::VerifyRequestData, types::PaymentsResponseData>
+    for Expresscheckout
+{
+}
+
+impl api::PaymentVoid for Expresscheckout {}
+
+impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsResponseData>
+    for Expresscheckout
+{
+    fn get_headers(
+        &self,
+        req: &types::PaymentsCancelRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        self.build_headers(req, connectors)
+    }
+
+    fn get_content_type(&self) -> &'static str {
+        self.common_get_content_type()
+    }
+
+    fn get_url(
+        &self,
+        req: &types::PaymentsCancelRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        let res = req
+            .response
+            .as_ref()
+            .map_err(|_| errors::ConnectorError::FailedToObtainIntegrationUrl)?;
+        if let types::PaymentsResponseData::TransactionResponse { resource_id, .. } = res {
+            if let types::ResponseId::ConnectorTransactionId(r) = resource_id {
+                Ok(format!(
+                    "{}{}{}{}",
+                    self.base_url(connectors),
+                    "v2/txns/",
+                    r,
+                    "/void/"
+                ))
+            } else {
+                Err(errors::ConnectorError::FailedToObtainIntegrationUrl).into_report()
+            }
+        } else {
+            Err(errors::ConnectorError::FailedToObtainIntegrationUrl).into_report()
+        }
+    }
+
+    fn get_request_body(
+        &self,
+        _req: &types::PaymentsCancelRouterData,
+    ) -> CustomResult<Option<String>, errors::ConnectorError> {
+        Ok(Some(String::from("")))
```

Comment:
This is not necessary to be defined as this is not being used in the request. There is a default implementation which returns an empty body